test/Fees.t.sol

```diff
@@ -0,0 +1,146 @@
+        PDPFees.proofFeeWithGasFeeBound(1, 0, 0, 1e18, epochs_per_day);
+    }
+
+    function testProofFeeWithGasFeeBoundZeroRawSize() public {
+        vm.expectRevert("failed to validate: raw size must be greater than 0");
+        PDPFees.proofFeeWithGasFeeBound(1, 5, 0, 0, epochs_per_day);
+    }
+
+    function testProofFeeWithGasFeeBoundHighGasFee() public pure {
+        uint64 filUsdPrice = 5;
+        int32 filUsdPriceExpo = 0;
+        uint256 rawSize = 1e18;
+
+        uint256 rewardPerPeriod = computeRewardPerPeriod(filUsdPrice, filUsdPriceExpo, rawSize);
+
+        uint256 gasLimitRight = (rewardPerPeriod * PDPFees.GAS_LIMIT_RIGHT_PERCENTAGE) / 100;
+
+        uint256 estimatedGasFee = gasLimitRight;
+
+        uint256 fee = PDPFees.proofFeeWithGasFeeBound(estimatedGasFee, filUsdPrice, filUsdPriceExpo, rawSize, epochs_per_day);
+
+        assertEq(fee, 0, "Fee should be 0 when gas fee is high");
+    }
+
+    function testProofFeeWithGasFeeBoundMediumGasFee() public pure {
+        uint64 filUsdPrice = 5;
+        int32 filUsdPriceExpo = 0;
+        uint256 rawSize = 1e18;
+
+        uint256 rewardPerPeriod = computeRewardPerPeriod(filUsdPrice, filUsdPriceExpo, rawSize);
+
+        uint256 gasLimitLeft = (rewardPerPeriod * PDPFees.GAS_LIMIT_LEFT_PERCENTAGE) / 100;
+        uint256 gasLimitRight = (rewardPerPeriod * PDPFees.GAS_LIMIT_RIGHT_PERCENTAGE) / 100;
+
+        uint256 estimatedGasFee = (gasLimitLeft + gasLimitRight) / 2;
+
+        uint256 fee = PDPFees.proofFeeWithGasFeeBound(estimatedGasFee, filUsdPrice, filUsdPriceExpo, rawSize, epochs_per_day);
+
+        uint256 expectedFee = gasLimitRight - estimatedGasFee;
+
+        assertEq(fee, expectedFee, "Fee should be partially discounted");
+    }
+
+    function testProofFeeWithGasFeeBoundLowGasFee() public pure {
+        uint64 filUsdPrice = 5;
+        int32 filUsdPriceExpo = 0;
+        uint256 rawSize = 1e18;
+
+        uint256 rewardPerPeriod = computeRewardPerPeriod(filUsdPrice, filUsdPriceExpo, rawSize);
+
+        uint256 gasLimitLeft = (rewardPerPeriod * PDPFees.GAS_LIMIT_LEFT_PERCENTAGE) / 100;
+
+        uint256 estimatedGasFee = gasLimitLeft / 2;
+
+        uint256 fee = PDPFees.proofFeeWithGasFeeBound(estimatedGasFee, filUsdPrice, filUsdPriceExpo, rawSize, epochs_per_day);
+
+        uint256 expectedFee = (rewardPerPeriod * PDPFees.PROOF_FEE_PERCENTAGE) / 100;
+
+        assertEq(fee, expectedFee, "Fee should be full proof fee when gas fee is low");
+    }
+
+    function testProofFeeWithGasFeeBoundNegativeExponent() public pure {
+        uint64 filUsdPrice = 5000;
+        int32 filUsdPriceExpo = -3;
+        uint256 rawSize = 1e18;
+        uint256 estimatedGasFee = 1e15;
+
+        uint256 fee = PDPFees.proofFeeWithGasFeeBound(estimatedGasFee, filUsdPrice, filUsdPriceExpo, rawSize, epochs_per_day);
+        assertTrue(fee > 0, "Fee should be positive with negative exponent");
+    }
+
+    function testProofFeeWithGasFeeBoundLargeRawSize() public pure {
```

Comment:
I don't understand the point of this test.

Reply:
Just a test with the biggest data size (1e30 bytes) among all the tests. Dosent hurt.